assembly/src/procedures/mod.rs

```diff
@@ -162,12 +162,14 @@ impl AsRef<str> for ProcedureName {
 impl Serializable for ProcedureName {
     fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
         let name_bytes = self.name.as_bytes();
-        let num_bytes = name_bytes.len() as u8;
-        if num_bytes > MAX_LABEL_LEN {
-            return Err(SerializationError::LengthTooLong);
-        }
+        let num_bytes = name_bytes.len();
+
+        debug_assert!(
+            num_bytes <= MAX_LABEL_LEN,
```

Comment:
I tried it, but `parse_label` takes ownership of the value, so this can be used only if we add an extra allocation. I don't have strong preferences here.

Reply:
I think that should be fine as we should wrap the check in the `debug_assert!(LABEL_PARSER...)` so the allocation will only take place in debug mode.

Reply:
changed